cmd/hlb/command/run.go

```diff
@@ -303,11 +308,83 @@ func ParseModuleURI(ctx context.Context, cln *client.Client, stdin io.Reader, ur
 		}
 		defer f.Close()
 		return parser.Parse(ctx, f)
+	case "git", "ssh":
+		return parseModuleGitURI(ctx, cln, uri)
 	default:
 		return nil, fmt.Errorf("%q is not a valid module uri scheme", u.Scheme)
 	}
 }
 
+func parseModuleGitURI(ctx context.Context, cln *client.Client, uri string) (*ast.Module, error) {
+	u, err := gitscheme.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+	if u.Filename == "" {
+		u.Filename = DefaultHLBFilename
+	}
+
+	var (
+		sessionOpts []llbutil.SessionOption
+		remote      string
+		root        string
+	)
+
+	sockPath := local.Env(ctx, "SSH_AUTH_SOCK")
+	if sockPath == "" {
+		root = u.Host + u.Path
+		remote = "https://" + root
```

Comment:
Confused about why this uses `https://`. Wouldn't the user-specified URL be either a `git://` or `ssh://` URL?

Reply:
This is HLB's git scheme, not the one specified by `git`. The idea is to provide a URI that can also specify branch and file path within a repo, and have logic to detect whether to use https vs ssh.

Other schemes could be something like `dockerfile://`, or like `gist://`. I didn't want the `https://` to arbitrarily mean git repo, but you're right in that `ssh://` breaks the pattern too.